src/formatting.rs

```diff
@@ -119,40 +157,86 @@ impl FormatTemplate {
 
         Ok(rendered)
     }
+}
 
-    // Experimental function: avoid using this function in your block.
-    // TODO reconsider the interface
-    pub fn render_widgets(
-        &self,
-        config: SharedConfig,
-        id: usize,
-        vars: &HashMap<&str, Value>,
-    ) -> Result<Vec<RenderedWidget>> {
-        let mut rendered = Vec::new();
-
-        for token in &self.tokens {
-            match token {
-                Token::Text(text) => rendered.push(RenderedWidget::Text(
-                    TextWidget::new(id, rendered.len(), config.clone()).with_text(text),
-                )),
-                Token::Var(var) => rendered.push(RenderedWidget::Var(
-                    var.name.clone(),
-                    TextWidget::new(id, rendered.len(), config.clone())
-                        .with_spacing(Spacing::Hidden)
-                        .with_text(
-                            &vars
-                                .get(&*var.name)
-                                .internal_error(
-                                    "util",
-                                    &format!("Unknown placeholder in format string: {}", var.name),
-                                )?
-                                .format(&var)?,
-                        ),
-                )),
+impl Default for FormatTemplate {
+    fn default() -> Self {
+        Self {
+            full: None,
```

Comment:
A simple `derive` is sufficient.

Reply:
Ah yes those are the parts of Rust that i don't have an intuition for yet